File: MaximumSubarray/maximum_subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        
        # Maximum so far
        maximum = nums[0]
        
        # Maximum ending here
        curr_max = nums[0]
        
        for i in range(1, len(nums)) :
            
            curr_max = max(nums[i], curr_max + nums[i])
            
            maximum = max(curr_max, maximum)
            
        return maximum

File: MaximumSubarray/maximum_subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    // Maximum so far
    let maximum = nums[0]

    // Maximum ending here
    let curr_max = nums[0]

    for (let index = 1; index < nums.length; index++) {
        curr_max = Math.max(nums[index], curr_max + nums[index]);
        
        maximum = Math.max(curr_max, maximum);
    }

    return maximum;

};